### DISCUSSION TITLE
System-prefixed paths in yaml.schemas

### DISCUSSION THREAD

**Discussion by amelio-vazquez-reina** [redhat-developer/vscode-yaml#751] [Q&A] [answered]
I currently have the following in `yaml.schemas` in one of my computers:
```
   "yaml.schemas": {
      "file:///Users/amelio/.vscode-insiders/extensions/atlassian.atlascode-2.3.1/resources/schemas/pipelines-schema.json": "bitbucket-pipelines.yml"
    },
```

which in a different computer is:
```
   "yaml.schemas": {
      "file:///Users/avazquezreina/.vscode-insiders/extensions/atlassian.atlascode-2.3.1/resources/schemas/pipelines-schema.json": "bitbucket-pipelines.yml"
    },
```

Note that the only difference is that my `$HOME` path (user name) is different (`amelio` vs `avazquezreina`). How can I make this system agnostic so that I can use the same `settings.json` file across two computers? 

E.g. Does this setting support environment variables?

**Comment by JPinkney** [comment, marked as answer]
I don't think VSCode supports having env variables in settings: https://github.com/microsoft/vscode/issues/2809

A workaround would be directly linking to the schema itself: https://bitbucket.org/atlassianlabs/atlascode/raw/4e9ef8faf727737cd9585bb01340117cd8236d44/resources/schemas/pipelines-schema.json

**Comment by Shaphil** [comment]
I am having the same issue while switching between Windows and Ubuntu. If the schema is updated regularly, hardcoding a link https://bitbucket.org/atlassianlabs/atlascode/raw/4e9ef8faf727737cd9585bb01340117cd8236d44/resources/schemas/pipelines-schema.json may pose problems. 
Can `yaml.schemas` be an array in that case, so that it can check if any of the values is correct, that will be used? Like this,

```json
"yaml.schemas": [
  { "file:///{linux_path}/atlassian.atlascode-2.8.6/resources/schemas/pipelines-schema.json": "bitbucket-pipelines.yml" },
  { "file:///{windows_path}/atlassian.atlascode-2.8.6/resources/schemas/pipelines-schema.json": "bitbucket-pipelines.yml" },
]
```

If on the other hand the schema does not change, setting the `yaml.schemas` value to a url by the extension itself would be nicer.
